components/net/fetch/methods.rs

```diff
@@ -866,6 +869,405 @@ async fn scheme_fetch(
     }
 }
 
+fn consider_local_directory_listing(
+    request: &mut Request,
+    url: ServoUrl,
+    file_path: std::path::PathBuf,
+) -> Response {
+    /*
+    If you want to be able to browse local directories, configure
+    Servo prefs so that "browser.local_directory_listing.enabled"
+    is set to true.
+    */
+    if pref!(browser.local_directory_listing.enabled) {
+        display_local_directory_listing(request, url, file_path)
+    } else {
+        Response::network_error(NetworkError::Internal(
+            "Local directory listing feature has not been enabled in preferences".into(),
+        ))
+    }
+}
+
+fn display_local_directory_listing(
+    request: &mut Request,
+    url: ServoUrl,
+    file_path: std::path::PathBuf,
+) -> Response {
+    if !request.origin.is_opaque() {
+        /*
+        Checking for an opaque origin may not fully guarantee that the request is safe
```

Comment:
I've simplified the comment, and added a TODO for creating a GitHub issue. I won't create the issue itself until this code gets accepted into Servo, otherwise the issue will be impossible for anyone else to take on. (If you want it raised anyway, before any merge, then let me know.)